core/spec.md

```diff
@@ -1263,6 +1379,72 @@ HTTP body.
 
 ---
 
+### `xid` Resolution API
+
+The resolution API is used to translate an `xid` into a URL that can be used
+to access an xRegistry entity. An implementation of the registry API SHOULD
+provide a resolution API under the path `/resolution`.
+The resolution API MAY also be offered independently of the registry API.
+It can resolve `xids` and federate access to multiple registries.
+In the following, `/` is the root of the resolution API, even if it would be
+provided under `/resolution` as part of an registry.
+
+#### `GET /xid/{xid}` and `HEAD /xid/{xid}`
+
+The placeholder `{xid}` depicts the `xid` to resolve. As `xids` MAY be full
+URIs or URI references, the `xid` MUST be percent-encoded accordingly.
+
+For `GET` and `HEAD` requests, the server MUST respond as follows:
+
+- If the `xid` is not found, the server MUST respond with an HTTP
+  `404 Not Found`.
+- If the `xid` is found and there exists exactly one URL to access the entity,
+  the server MUST respond with an HTTP `308` where the `Location` header
+  MUST contain the URL to access the entity.
+- If the `xid` is found and there are multiple URLs to access the entity, the
+  server MUST respond with an HTTP `300 Multiple Choices` where the choices
```

Comment:
Like above, I'm not sure a 300 (or 3xx in general) is correct since I don't think we're doing a "redirect" in the HTTP sense. I think we're calling a function and the result is one or more URLs - that's not a redirect, that's a response. We can then choose how the response looks (headers vs body) in the single vs multiple cases.

Reply:
Well, we can discuss this. I thought it might be neat if even a standard web browser could be used for this. I could imagine returning 200 and the JSON in case of a GET. We could then do the redirect only for the HEAD call. Behaving like a redirect by returning the according headers without actually being one looks strange to me. If we use 30x, a web browser would automatically retrieve the entity. Someone who is aware of xRegistry, could still decide to ignore the 30x. 
The redirect approach only works for queries that specify an `xid`. For calls that might return data sets for multiple xids, a redirect doesn't make sense.